pkg/repo/chartrepo.go

```diff
@@ -182,28 +184,29 @@ func (r *ChartRepository) generateIndex() error {
 
 // FindChartInRepoURL finds chart in chart repository pointed by repoURL
 // without adding repo to repositories
-func FindChartInRepoURL(repoURL, chartName, chartVersion, certFile, keyFile, caFile string, getters getter.Providers) (string, error) {
-	return FindChartInAuthRepoURL(repoURL, "", "", chartName, chartVersion, certFile, keyFile, caFile, getters)
+func FindChartInRepoURL(repoURL, chartName, chartVersion, certFile, keyFile, caFile string, renegotiate string, getters getter.Providers) (string, error) {
```

Comment:
Because this change modifies the function signature, this would be considered a backwards compatibility breaker.

A more in-depth refactor may be necessary here for this area of the code base. Perhaps something like the [builder pattern](https://en.wikipedia.org/wiki/Builder_pattern) may be useful here. This is how `getter`s can introduce new features like `WithTLSRenegotiate` without breaking existing use cases.

What do you think?

Reply:
I refactored the whole code in a way that I think makes sense but again I'm very new to Golang so this might not be what you meant with your comment, I'm happy to take director or get feedback.